classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -659,79 +609,109 @@ protected function renderAssociations($object, $depth, $associations, $ws_params
                 if (isset($this->wsResource[$assoc_name]['class']) && class_exists($this->wsResource[$assoc_name]['class'], true)) {
                     $class_name = $this->wsResource[$assoc_name]['class'];
                 }
-                $output_details = '';
-                foreach ($objects_assoc as $object_assoc) {
-                    if ($depth == 0 || $class_name === null) {
-                        $value = null;
-                        if (!empty($object_assoc)) {
-                            $value = $object_assoc;
-                        }
-                        if (empty($fields_assoc)) {
-                            $fields_assoc = [['id' => $value['id']]];
-                        }
-                        $output_details .= $this->renderFlatAssociation($object, $depth, $assoc_name, $association['resource'], $fields_assoc, $value, $parent_details);
-                    } else {
-                        foreach ($object_assoc as $id) {
-                            if ($class_name !== null) {
-                                $child_object = new $class_name($id);
-                                $output_details .= $this->renderEntity($child_object, ($depth - 2 ? 0 : $depth - 2));
-                            }
-                        }
-                    }
-                }
-                if ($output_details != '') {
-                    $output .= $this->setIndent($depth) . $this->objectRender->renderAssociationHeader($object, $ws_params, $assoc_name);
-                    $output .= $output_details;
-                    $output .= $this->setIndent($depth) . $this->objectRender->renderAssociationFooter($object, $ws_params, $assoc_name);
-                } else {
-                    $output .= $this->setIndent($depth) . $this->objectRender->renderAssociationHeader($object, $ws_params, $assoc_name, true);
-                }
+
+                $this->injectAssociation($associationsWrapperNode, $object, $assoc_name, $association, $ws_params, $class_name, $objects_assoc, $fields_assoc);
+            }
+        }
+    }
+
+    /**
+     * Inject one association section into parent node
+     *
+     * @param ApiNode $parentNode
+     * @param string $assoc_name
+     * @param array $params
+     *
+     * @return void
+     */
+    private function injectAssociation(&$parentNode, $object, $assoc_name, $association, $params, $class_name, $objects_assoc, $fields_assoc)
```

Comment:
I'm bit unsure here. My imlpementation hereis not using $class_name parameter in here, which could lead to errors. Could please someone point to a situation where $class_name has been used, in previous version - so I can test it and upgrade it?